pkg/controller/install/webhook.go

```diff
@@ -162,6 +182,71 @@ func (i *StrategyDeploymentInstaller) createOrUpdateValidatingWebhook(ogNamespac
 			return err
 		}
+
+		if len(desc.ConversionCrds) != 0 {
+			if err := createOrUpdateConversionCrds(desc, webhook.Webhooks[0].ClientConfig, i); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func createOrUpdateConversionCrds(desc v1alpha1.WebhookDescription, clientConfig admissionregistrationv1.WebhookClientConfig, i *StrategyDeploymentInstaller) error {
+
+	for index, ConversionCrd := range desc.ConversionCrds {
+		crd, err := i.strategyClient.GetOpLister().APIExtensionsV1().CustomResourceDefinitionLister().Get(desc.ConversionCrds[index])
+
+		// check if CRD is present on cluster
+		if err != nil {
+			log.Info("CRD not found %s, error: %s", desc.ConversionCrds[index], err.Error())
+		}
+
+		log.Info("Found CRD %s", desc.ConversionCrds[index])
+
+		// check if the CSV owns this CRD
+		foundCRD := false
+		csv, err := i.strategyClient.GetOpLister().OperatorsV1alpha1().ClusterServiceVersionLister().ClusterServiceVersions(i.owner.GetNamespace()).Get(i.owner.GetName())
+		ownedCRDs := csv.Spec.CustomResourceDefinitions.Owned
+		for _, ownedCRD := range ownedCRDs {
+			if ownedCRD.Name == crd.GetName() {
+				log.Info("CSV %q owns CRD %q", csv.GetName(), crd.GetName())
+				foundCRD = true
+			}
+		}
+
+		// crd.Spec.Conversion.Strategy specifies how custom resources are converted between versions.
+		// Allowed values are:
+		// 	- None: The converter only change the apiVersion and would not touch any other field in the custom resource.
+		// 	- Webhook: API Server will call to an external webhook to do the conversion. This requires crd.Spec.preserveUnknownFields to be false.
+		// References:
+		//  - https://docs.openshift.com/container-platform/4.5/rest_api/extension_apis/customresourcedefinition-apiextensions-k8s-io-v1.html
+		// 	- https://kubernetes.io/blog/2019/06/20/crd-structural-schema/#pruning-don-t-preserve-unknown-fields
+		if (crd.Spec.Conversion.Strategy == "Webhook" || crd.Spec.Conversion.Strategy == "None") && foundCRD {
+			if crd.Spec.Conversion.Strategy == "Webhook" && crd.Spec.PreserveUnknownFields == true {
+				log.Info("crd.Spec.PreserveUnknownFields must be false to let API Server call webhook to do the conversion.")
+			} else {
+				log.Info("updating CRD")
+				crd.Spec.Conversion = &apiextensionsv1.CustomResourceConversion{
+					Webhook: &apiextensionsv1.WebhookConversion{
+						ClientConfig: &apiextensionsv1.WebhookClientConfig{
+							Service: &apiextensionsv1.ServiceReference{
+								Namespace: clientConfig.Service.Namespace,
+								Name:      clientConfig.Service.Name,
+								Port:      clientConfig.Service.Port,
+							},
+							CABundle: clientConfig.CABundle,
+						},
+						ConversionReviewVersions: desc.AdmissionReviewVersions,
+					},
+				}
+			}
+
+			if _, err = i.strategyClient.GetOpClient().ApiextensionsInterface().ApiextensionsV1().CustomResourceDefinitions().Update(context.TODO(), crd, metav1.UpdateOptions{}); err != nil {
```

Comment:
Do we care at all about which client we are using to update? `v1` versus `v1beta1`?

Reply:
we should generally pick the latest one available on the cluster, so v1 seems right